### DISCUSSION TITLE
A way to retrieve scenes instanced as a tile on a tilemap

### DISCUSSION THREAD

**Discussion by feelingsonice** [godotengine/godot-proposals#6422] [2D]
So Godot 4 now supports [adding scenes](https://docs.godotengine.org/en/latest/tutorials/2d/using_tilesets.html#using-a-collection-of-scenes) to a TileMap as a tile. It only makes sense to add a way to retrieve the scenes instanced on a specific coordinate, so that the user can use it like an actual scene by calling methods available on it.

**Comment by Calinou** [comment]
This may be feasible already (cc @groud) using a combination of `local_to_map()` methods and a TileSet method to get a tile data according to its ID. If that's the case, it's a matter of documentation :slightly_smiling_face:

**Reply by thommor** [reply]
Have you got an example of how you would achieve this?

**Reply by Calinou** [reply]
Sorry, I don't know how to achieve this.

**Comment by feelingsonice** [comment]
@Calinou could you be so kind to advise a work-around for the time-being?

**Reply by AThousandShips** [reply]
Please read this: https://github.com/godotengine/godot-proposals/discussions/6422#discussioncomment-5654204

Someone else might know though

**Reply by feelingsonice** [reply]
I saw that, but I feel that this is a major hindrance to the new feature. Placing scenes onto tilemaps would help simplify a lot of but I need a way to retrieve the ones instanced. Would love to know if someone knows.

**Reply by AThousandShips** [reply]
I'm sorry I assumed you hadn't because you asked someone who said they didn't know how to do it

**Reply by AThousandShips** [reply]
There isn't a straight forward way to get the actual instanced scenes, but the ability to get the packed scene of a specific tile can be done with some combination of [`TileMap.get_cell_atlas_coords`](https://docs.godotengine.org/en/stable/classes/class_tilemap.html#class-tilemap-method-get-cell-atlas-coords), and [`TileSetScenesCollectionSource.get_scene_tile_scene`](https://docs.godotengine.org/en/stable/classes/class_tilesetscenescollectionsource.html#class-tilesetscenescollectionsource-method-get-scene-tile-scene)

Direct support for accessing the scenes themselves isn't available, and the instanced scenes are considered internal, they might also be reinstanced if the quadrant is updated, unsure

**Reply by AThousandShips** [reply]
* Related, see issues raised there: #4833

**Reply by feelingsonice** [reply]
Interesting. Could we consider adding this in future updates? Feel like it'd be a feature people would like.

**Reply by AThousandShips** [reply]
Thats what that proposal is for, but there's still the issues raised there to consider for this discussion as well, and I agree with groud about that exposing access to these internal scenes is something that might not be desirable, and see the suggestions for workarounds there
